ui/app/src/context/SnackbarProvider.tsx

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useCallback } from 'react';
+import {
+  SnackbarProvider as NotistackProvider,
+  ProviderContext as NotistackContext,
+  useSnackbar as useNotistack,
+  SnackbarMessage,
+  OptionsObject,
+  SnackbarKey,
+} from 'notistack';
+
+export interface SnackbarContext extends NotistackContext {
+  errorSnackbar: EnqueueFunction;
+  infoSnackbar: EnqueueFunction;
+  warningSnackbar: EnqueueFunction;
+  successSnackbar: EnqueueFunction;
+
+  /**
+   * Useful for catch blocks where the error will be of type `unknown`, tries
+   * to show the `message` property if passed an instance of `Error`.
+   */
+  exceptionSnackbar: (error: unknown, options?: SnackbarOptions) => SnackbarKey;
+}
+
+type EnqueueFunction = (
+  message: SnackbarMessage,
+  options?: SnackbarOptions
+) => SnackbarKey;
+
+type SnackbarOptions = Omit<OptionsObject, 'variant'>;
+
+/**
+ * Application-wide provider for showing snackbars/toasts.
+ */
+export const SnackbarProvider = NotistackProvider;
+
+/**
+ * Gets the SnackbarContext with methods for displaying snackbars/toasts.
+ */
+export function useSnackbar(): SnackbarContext {
+  const { enqueueSnackbar, closeSnackbar } = useNotistack();
+
+  // Create variant-specific callbacks
+  const errorSnackbar = useEnqueueFunction(enqueueSnackbar, 'error');
+  const infoSnackbar = useEnqueueFunction(enqueueSnackbar, 'info');
+  const warningSnackbar = useEnqueueFunction(enqueueSnackbar, 'warning');
+  const successSnackbar = useEnqueueFunction(enqueueSnackbar, 'success');
```

Comment:
We've found these super useful for a couple reasons:

1. We often want the ability to potentially show different types of snackbars in a single page/hook, so being able to grab what you need (including multiple) is nice, e.g.:
```ts
const { successSnackbar, exceptionSnackbar } = useSnackbar();
```
2. We almost never need to create and pass an options object which tends to look a lot less verbose. So instead of:
```ts
enqueueSnackbar('Something really bad happened', { 
  variant: 'error'
});
```
We can just do:
```ts
errorSnackbar('Something really bad happened');
```